src/nri/sampler.go

```diff
@@ -72,16 +58,18 @@ func NewSampler(hostRoot, cgroupPath string) (ContainerSampler, error) {
 // SampleAll populates the integration of the argument with metrics and labels from all the containers in the system
 // running and non-running
 func (cs *ContainerSampler) SampleAll(i *integration.Integration) error {
-	// todo: configure to retrieve only the running containers
-	containers, err := cs.docker.ContainerList(context.Background(), types.ContainerListOptions{All: true})
-	if err != nil {
-		return err
-	}
 	defer func() {
 		if err := cs.store.Save(); err != nil {
 			log.Warn("persisting previous metrics: %s", err.Error())
 		}
 	}()
+
+	// todo: configure to retrieve only the running containers
+	containers, err := cs.docker.ContainerList(context.Background(), types.ContainerListOptions{All: true})
```

Comment:
could we generate ctx in main? then pass it & handle properly?

Reply:
I'm ok with generating the context in main.

You mean handling cancellation? If yes, I can this can be subject of another PR to avoid extra complications and increased scope here.